include/zephyr/net/conn_mgr.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_CONN_MGR_H_
+#define ZEPHYR_INCLUDE_NET_CONN_MGR_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(CONFIG_NET_CONNECTION_MANAGER)
+
+struct net_if;
+struct net_l2;
+
+/**
+ * @brief Resend either NET_L4_CONNECTED or NET_L4_DISCONNECTED depending on whether connectivity
+ * is currently available.
+ */
+void net_conn_mgr_resend_status(void);
+
+/**
+ * @brief Mark an iface to be ignored by conn_mgr.
+ *
+ * @param iface - iface to be ignored.
+ */
+void net_conn_mgr_ignore_iface(struct net_if *iface);
```

Comment:
I think `*_watch` and `*_unwatch` could also work.

Should we explain what is the implication of ignoring an interface or it's just too obvious?

Reply:
I don't really have a strong preference, but `net_conn_mgr_monitor_iface` could also be an option (with an extra bool parameter to enable/disable).

Reply:
I don't have a strong preference on naming either, though I lean towards "ignore/unignore" simply because it suggests that ifaces are paid attention by default (which they are). @wbober Since Robert doesn't have a strong opinion, and you raised the issue originally, which of the three options would be your final choice?

I agree that the documentation for these functions needs to be improved. I will make sure to do that

Reply:
I've improved the specificity of the documentation, although the behavior described technically requires a later commit to be included. I feel that is a lesser evil than rewriting the docs in the commit that changes the behavior. I will include a note about it in the commit message.